maven-core/src/main/java/org/apache/maven/lifecycle/internal/DependencyContext.java

```diff
@@ -48,7 +43,7 @@ public class DependencyContext {
 
     private final Collection<String> scopesToResolveForAggregatedProjects;
 
-    private volatile Collection<?> lastDependencyArtifacts = UNRESOLVED;
+    private volatile Collection<?> lastDependencyArtifacts = new ArrayList<>();
```

Comment:
`Collections.emptyList()`, as this collection is never written to.

Reply:
fixed